### DISCUSSION TITLE
Write your own AutoMapper in C#

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#49] [General]
# Write your own AutoMapper in C#

Sometimes you have to map an object to another representation in C#. And you think: Why isn't C# duck-typing capable?
You might hear of libraries like AutoMapper that do the tedious work of mapping one object to another with the same structure. This blog post will give a super simple introduction to how those libraries are working internally.
In the end a bit of a subjective topic on whether or not I would use such libraries.

https://steven-giesel.com/blogPost/16bf84c2-1464-40ab-8ef6-58f225e21c15

**Comment by dineshmike** [comment]
Mapperly is performant alternative to AutoMapper which uses Source Generator. 
[Link](https://github.com/riok/mapperly)

**Reply by linkdotnet** [reply]
Nice. That for example is a bit more explicit! I do like that approach much better than AutoMapper. Any experience with that library @dineshmike ?

**Reply by latonz** [reply]
I'm one of the authors of Mapperly. I appreciate any feedback and I'm happy to answer any questions!

**Reply by linkdotnet** [reply]
After a year I do manage to answer @latonz - first of all I do like the library (even though I am personally still not in favor of using 3rd party libraries, but that is just me.)

I even covered that in another blog-post: https://steven-giesel.com/blogPost/3368bf96-648e-4fa2-9c6c-be71e40ac4cc/an-alternative-to-automapper 
As I do feel your library is in the perfect middle ground of those two worlds!

So cheers to you!

**Comment by fabiogalante** [comment]
Read this article

https://medium.com/@samuilovas/c-mappers-which-one-is-the-fastest-17ac6889ce4f

Reflection is the slowest of all

**Comment by jvmlet** [comment]
I'm sorry to tell all AutoMapper fans: save your time and stop using it.  
Mapster is much better choice with it's build time mapping generation feature and build time  mapping validation.
Start using strongly typed  generated mapping interfaces and avoid runtime surprises with AutoMapper generic API.
Kudos to Mapster authors !!!
